### PR TITLE
Autodeploy website with Github Actions

### PR DESCRIPTION
**Pull request by Vinetos**
Discussed in #39

### CODE REVIEW

**Comment by Vinetos** [comment]
> Hi @Vinetos, this seems good!
>
> I'm not an expert in GitHub actions, but by reading the file you proposed, It does not compile the TypeScript code to pure JavaScript, right?
> We should try to do that too!

Yes it does.  

```yml
      - name: Build
        run: |
          npm clean-install
          npm run build
```
This part build the project. It runs `npm clean-install` and then  `npm run build`.  

The second command generates the website ready to be released. It creates a folder at the root of the project called `dist/` and "compile" the project to generate `.html, .js, .css` and assets folder. 

Here is a screen after running `npm run build` :
image

Then, the third step is to deploy this folder to the organisation folder. I use `JamesIves/github-pages-deploy-action` which automate the deployment to the organisation project.

```yml 
      # Deploy
      - name: Deploy
        uses: JamesIves/github-pages-deploy-action@releases/v3
        with:
          # Where the compiled files should be pushed
          REPOSITORY_NAME: tableaunoir/tableaunoir.github.io
          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
          # On which branch. Sometimes, people use the gh-pages branch of a project to host their projects.
          BRANCH: master
          # Where are the compiled files (pure js, html, etc.)
          FOLDER: dist
          CLEAN: true
```

To resume, this workflow :
- builds the project
- push the contents of `dist` folder to `tableaunoir/tableaunoir.github.io` on `master` branch

**Comment by Naereen** [comment]
Oh awesome @Vinetos! I read this line, and wasn't sure if `npm build` was enough to run the tsc TS compiler!
Great! So I greatly approve this PR!

**Comment by Vinetos** [comment]
Yep it is ! 
I am going to mark this PR as WIP because there is bad behaviour which build and deploy the for each version of npm in `node-version: [10.x, 12.x, 14.x]`. 
Maybe should we only use one version for deployment.

**Comment by Vinetos** [comment]
This PR is now ready to be reviewed ! 😄

**Comment by francoisschwarzentruber** [comment]
Hello Vinetos, thank you very much for your work.

**Comment by francoisschwarzentruber** [comment]
Very funny. I received a notification entitled "Run failed: .github/workflows/deploy.yml" whose contents is ".github/workflows/deploy.yml: All jobs were successful". ;)

**Comment by francoisschwarzentruber** [comment]
In the website, the following error message appears:

> 
>  Error
>  every step must define a `uses` or `run` key

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    runs-on: ubuntu-latest
+    
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - name: Using Node.js ${{ matrix.node-version }}
+      - uses: actions/checkout@v2
```
Comment (Vinetos) [inline .github/workflows/deploy.yml:24]: I added a typo when I copy/pasted ! It should be
` uses: actions/checkout@v2`

Comment (Vinetos) [reply .github/workflows/deploy.yml:24]:
```suggestion
        uses: actions/checkout@v2
```

**Comment by Vinetos** [comment]
I'll resolve theses bugs in a new pull request

**Pull request merged**